airgun/views/subscription.py

```diff
@@ -104,6 +114,12 @@ def is_displayed(self):
         return self.browser.wait_for_element(
             'div#subscriptions-table', timeout=10, exception=False) is not None
 
+    def search(self, query):
+        """Customized search to make sure that the table spinner is hidden, after search"""
+        self.searchbox.search(query)
+        wait_for_spinner(self)
```

Comment:
same here. We start to add special methods just for waiter. We need to move that logic on a level above in our hierarchy